animal-options.lua

```diff
@@ -0,0 +1,175 @@
+    if unit.sex ~= 1 or unit.flags3.gelded == true then
+        return false
+    else
+        return true
+    end
+end
+
+creature_screen=defclass(creature_screen, overlay.OverlayWidget)
+creature_screen.ATTRS {
+    desc = "Add options to tamed animals view sheet",
+    default_pos={x=-42,y=37},
+    default_enabled=false,
+    viewscreens='dwarfmode/ViewSheets/UNIT/Overview',
+    frame={w=23, h=6},
+}
+
+-- The above function already handles checking if valid unit
+-- so just set slaughter flag
+function creature_screen:set_slaughter_flag(option)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if not unit then return end
+
+    if option == YES then
+        unit.flags2.slaughter = true
+    else
+        unit.flags2.slaughter = false
+    end
+end
+
+-- set units marked for gelding flag
+function creature_screen:set_geld_flag(option)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if not unit then return end
+
+    if option == YES then
+        unit.flags3.marked_for_gelding = true
+    else
+        unit.flags3.marked_for_gelding = false
+    end
+end
+
+-- set available for adoption flag
+function creature_screen:set_adoption_flag(option)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if not unit then return end
+
+    if option == YES then
+        unit.flags3.available_for_adoption = true
+    else
+        unit.flags3.available_for_adoption = false
+    end
+end
+
+-- Check current flag status of animal to dynamically set on/off
+function creature_screen:get_butcher(unit)
+    if not unit then return end
+
+    if unit.flags2.slaughter == true then
+        return YES
+    else
+        return NO
+    end
+end
+
+function creature_screen:get_geld(unit)
+    if not unit then return end
+
+    if unit.flags3.marked_for_gelding == true then
+        return YES
+    else
+        return NO
+    end
+end
+
+function creature_screen:get_adopt(unit)
+    if not unit then return end
+
+    if unit.flags3.available_for_adoption == true then
+        return YES
+    else
+        return NO
+    end
+end
+
+-- Use render to set On/Off dynamically for each unit
+function creature_screen:render(dc)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    self.subviews.butcher_animal:setOption(self:get_butcher(unit))
+    self.subviews.geld_animal:setOption(self:get_geld(unit))
+    self.subviews.adopt_animal:setOption(self:get_adopt(unit))
+
+    creature_screen.super.render(self, dc)
+end
+
+function creature_screen:init()
+self:addviews{
+    widgets.Panel{
+        frame_background=gui.CLEAR_PEN,
+        frame_style=gui.FRAME_MEDIUM,
+        visible=check_valid_unit,
+        subviews={
+            widgets.CycleHotkeyLabel{
+                frame={t=0,l=0},
+                label='Butcher',
+                key='CUSTOM_CTRL_B',
+                options={
+                    {label=NO, value=NO, pen=COLOR_WHITE},
+                    {label=YES, value=YES, pen=COLOR_GREEN},
```

Comment:
```suggestion
                    {label=NO, value=false, pen=COLOR_WHITE},
                    {label=YES, value=true, pen=COLOR_GREEN},
```

Reply:
you can use `true` and `false` directly here as the option values and avoid the need to translate in each of the "set" functions.

also, yellow might be more appropriate than green for the "On" setting. Green means more "this is the good/safe setting", which isn't exactly applicable here

Reply:
Made colors yellow and switched values with booleans.